### PR TITLE
Reset mocks automatically after each test

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Motivation and Context

I don't like having to cleanup mocks manually after each call so I fixed it.

### CODE REVIEW

**Comment by PavelVanecek** [comment]
Hm the mocks are still not completely independent. I suspect that vitest runs things in parallel, and mocking global prototype will never be compatible with that.

**Comment by PavelVanecek** [comment]
I have an idea how to fix that. In the next PR. The changes from here are still useful so please consider merging it too! I will build on top.

File: test/component/__snapshots__/Legend.spec.tsx.snap [hunk 1]
```diff
@@ -633,7 +633,7 @@ exports[`<Legend /> > content as a function > should pass parameters to the func
             0,
             -Infinity,
           ],
-          "height": 290,
+          "height": 270,
```
Comment (ckifer) [inline test/component/__snapshots__/Legend.spec.tsx.snap:636]: Aren't these the wrong values?

**Review by ckifer — APPROVED** [review]
If you say this is good to go I trust you - a bit confused why the values changed

**Comment by PavelVanecek** [comment]
I didn't change the implementation therefore it's not that. Some of the tests fail when run together and pass when run individually which means the mocks affect each other. I'll get a better solution.

**Pull request merged**